crates/holochain_conductor_config/src/generate.rs

```diff
@@ -0,0 +1,101 @@
+//! Helpers for generating new directories and `ConductorConfig`.
+
+use std::path::PathBuf;
+
+use holochain_conductor_api::conductor::paths::{ConfigRootPath, KeystorePath};
+use kitsune_p2p_types::config::KitsuneP2pConfig;
+
+use crate::config::create_config;
+use crate::config::write_config;
+use crate::msg;
+use crate::ports::set_random_admin_port;
+
+/// Generate configurations
+/// This creates a directory containing a `ConductorConfig`,
+/// a keystore, and a database root directory.
+pub fn generate(
```

Comment:
It looks like there's always a `generate` followed by an `set_admin_port`. How about making the admin port a parameter of `generate` instead? Or even of `create_config`?